src/probability/conditional.lean

```diff
@@ -0,0 +1,140 @@
+for now, but the alternative theorems can be added if needed.
+
+Use of `@[simp]` generally follows the rule of removing conditions on a measure
+when possible.
+
+Hypotheses that are used to "define" a conditional distribution by requiring that
+the conditioning set has non-zero measure should be named using the abbreviation
+"c" (which stands for "conditionable") rather than "nz". For example `(hci : μ (s ∩ t) ≠ 0)`
+(rather than `hnzi`) should be used for a hypothesis ensuring that `μ[|s ∩ t]` is defined.
+
+## Tags
+conditional, conditioned, bayes
+-/
+
+noncomputable theory
+
+open measure_theory measurable_space
+
+variables {α : Type*} {m : measurable_space α} (μ : measure α) {s t : set α}
+
+namespace probability_theory
+
+section definitions
+
+/-- The conditional probability measure of measure `μ` on set `s` is `μ` restricted to `s` 
+and scaled by the inverse of `μ s` (to make it a probability measure):
+`(μ s)⁻¹ • μ.restrict s`. -/
+def cond (s : set α) : measure α :=
+  (μ s)⁻¹ • μ.restrict s
+
+end definitions
+
+local notation  μ `[` s `|` t `]` := cond μ t s
+local notation  μ `[|`:60 t`]` := cond μ t
+
+/-- The conditional probability measure of any finite measure on any set of positive measure
+is a probability measure. -/
+lemma cond_is_probability_measure [is_finite_measure μ] (hcs : μ s ≠ 0) :
+  is_probability_measure $ μ[|s] :=
+⟨by { rw [cond, measure.smul_apply, measure.restrict_apply measurable_set.univ,
+  set.univ_inter], exact ennreal.inv_mul_cancel hcs (measure_ne_top _ s) }⟩
+
+section bayes
+
+@[simp] lemma cond_univ [is_probability_measure μ] :
+  μ[|set.univ] = μ :=
+by simp [cond, measure_univ, measure.restrict_univ]
+
+/-- The axiomatic definition of conditional probability derived from a measure-theoretic one. -/
+@[simp] lemma cond_measure_apply (hms : measurable_set s) (t : set α) :
+  μ[t|s] = (μ s)⁻¹ * μ (s ∩ t) :=
+by { rw [cond, measure.smul_apply, measure.restrict_apply' hms, set.inter_comm], refl }
+
+lemma inter_pos_of_cond_ne_zero (hms : measurable_set s) (hcst : μ[t|s] ≠ 0) :
+  0 < μ (s ∩ t) :=
+begin
+  refine pos_iff_ne_zero.mpr (right_ne_zero_of_mul _),
+  { exact (μ s)⁻¹ },
+  convert hcst,
+  simp [hms, set.inter_comm]
+end
+
+variable [is_finite_measure μ]
+
+lemma cond_pos_of_inter_ne_zero (hms : measurable_set s) (hci : μ (s ∩ t) ≠ 0) :
+  0 < μ[|s] t :=
+begin
+  rw cond_measure_apply _ hms,
+  refine ennreal.mul_pos _ hci,
+  exact ennreal.inv_ne_zero.mpr (measure_ne_top _ _),
+end
+
+/-- Conditioning first on `s` and then on `t` results in the same measure as conditioning
+on `s ∩ t`. -/
+@[simp] lemma cond_cond_eq_cond_inter
+  (hms : measurable_set s) (hmt : measurable_set t) (hcs : μ s ≠ 0) (hci : μ (s ∩ t) ≠ 0) :
+  μ[|s][|t] = μ[|s ∩ t] :=
+begin
+  apply measure.ext, 
+  intros,
+  haveI := cond_is_probability_measure μ
+    (μ.to_outer_measure.pos_of_subset_ne_zero (set.inter_subset_left _ _) hci).ne',
+  simp [*, measure_ne_top, ennreal.mul_inv],
```

Comment:
This looks like a non-terminal simp (simp that does not close a goal). That can make the proof break in the future if the set of lemmas used by simp changes, therefore we avoid them. Can you replace that with `squeeze_simp` in your code, which will suggest something of the form `simp only [...]`, then use that expression?

If it's not a non-terminal simp but a sub-goal, then it needs `{}` around it.

Reply:
Ah, never thought of that. Thank you for the explanation!